server/rollback_runner.go

```diff
@@ -0,0 +1,160 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common/log"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/bytebase/bytebase/plugin/db/mysql"
+	"github.com/bytebase/bytebase/store"
+)
+
+// NewRollbackRunner creates a new rollback runner.
+func NewRollbackRunner(store *store.Store, dataDir string) *RollbackRunner {
+	return &RollbackRunner{
+		store:          store,
+		dataDir:        dataDir,
+		generateSignal: make(chan struct{}, 1),
+	}
+}
+
+// RollbackRunner is the rollback runner generating rollback SQL statements.
+type RollbackRunner struct {
+	store          *store.Store
+	dataDir        string
+	generateSignal chan struct{}
+	generateMap    sync.Map
+}
+
+// Run starts the rollback runner.
+func (r *RollbackRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	r.retryGenerateRollbackSQL(ctx)
+	for {
+		select {
+		case <-r.generateSignal:
```

Comment:
This is still blocking if the previous work isn't yet completed. Consider getting rid of the channel and use timer.

Reply:
Yes. The runner is blocking on receiving a signal and process all the SQL generation in the sync.Map.

A Timer is simpler here. I'll refactor to use a Timer instead.